Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Hole_filling/Triangulate_hole_polyline.h

```diff
@@ -254,14 +254,14 @@ class Weight_min_max_dihedral_and_area
       double angle = 0;
       // check whether the edge is border
       if( (v0 + 1 == v1 || (v0 == n-1 && v1 == 0) ) && !Q.empty() ) {
-        angle = 180 - CGAL::abs( 
+        angle = CGAL_PI - CGAL::abs( 
            to_double(CGAL::approximate_dihedral_angle(P[v0],P[v1],P[v_other],Q[v0])) );
       }
       else {
         if(e == 2) { continue; }
         if(lambda.get(v0, v1) != -1){
           const Point_3& p01 = P[lambda.get(v0, v1)];
-          angle = 180 - CGAL::abs( 
+          angle = CGAL_PI - CGAL::abs( 
```

Comment:
I'm surprised that there is no need of further changes since angle has changed units.

Reply:
Because later angles are only compared that are either all radians or all degree, but never to a number such as 180 or pi.

Reply:
I repeat that it means the code is badly written and should compare cosines and sines, instead of comparing angles.

I repeat here again not to say that you, Andreas, should try to modify and fix the code, but I repeat as many time as possible, to spread the idea, and hoping that next time the package writers will remember I said that, and will consult me before comparing angles.

A function computing angles should only be used if one needs to *display* angles to users, as statistics for example.

Reply:
I suggest that you make an issue for this.

Reply:
You are right. https://github.com/CGAL/cgal/issues/3630